splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt

```diff
@@ -13,16 +17,33 @@ class SplashActivity : BaseViewModelActivity<SplashViewModel>() {
     override val viewModelClass: KClass<SplashViewModel>
         get() = SplashViewModel::class
 
-    override fun onViewModelCreated(viewModel: SplashViewModel) {
-        super.onViewModelCreated(viewModel)
-        val handler = Handler()
+    override fun onResume() {
+        super.onResume()
+        viewModel.percentage
+                .map { "Percentage : ${it * 100}" }
+                .bindToMain(percentText.text())
+                .disposedBy(disposeBag)
 
-        /**
-         * TODO:: Remove the post delay once splash is functional.
-         */
-        handler.postDelayed({
-            val intent = Intent(this, MainActivity::class.java)
-            startActivity(intent)
-        }, 1000)
+        viewModel.navigateTo
+                .subscribe {
+                    when (it) {
+                        is Navigate.Forward -> {
+                            when (it.endpoint) {
+                                is SplashViewModel.NavigationEndpoint.Welcome -> {
+                                    startActivity(Intent(this@SplashActivity, MainActivity::class.java))
+                                    finish()
+                                }
+                            }
+                        }
+                        is Navigate.Back -> {
+
+                        }
+                    }
+                }.disposedBy(disposeBag)
+    }
+
+    override fun onPause() {
+        super.onPause()
+        disposeBag.clear()
```

Comment:
maybe we have a separate `navDisposeBag` that we use instead for bindings that occur only when screen is visible? in that case we probably want it in `onStop()/onStart()` so in multiwindow.

Reply:
hmm i guess we can do that.